### PR TITLE
Added paymentMethod DTO for /payments call

### PR DESCRIPTION
**Pull request by lancergr**
**Description**
Added paymentMethod DTO for /payments call

**Tested scenarios**
/payments with CC and APM

### CODE REVIEW

File: src/main/java/com/adyen/model/checkout/PaymentsRequest.java [hunk 1]
```diff
@@ -639,26 +615,26 @@ public PaymentsRequest addEncryptedCardData(String encryptedCardNumber, String e
...
         return this;
     }
 
     public PaymentsRequest addOneClickData(String recurringDetailReference, String encryptedSecurityCode) {
-        this.paymentMethod = new HashMap<>();
-        this.paymentMethod.put(METHOD_TYPE, TYPE_SCHEME);
-        this.paymentMethod.put(RECURRING_DETAIL_REFERENCE, recurringDetailReference);
-        this.paymentMethod.put(ENCRYPTED_SECURITY_CODE, encryptedSecurityCode);
+        DefaultPaymentMethodDetails paymentMethodDetails = new DefaultPaymentMethodDetails();
+        this.paymentMethod = paymentMethodDetails;
```
Comment (rkewlani) [inline src/main/java/com/adyen/model/checkout/PaymentsRequest.java]: minor: exchange of 634 and 637 will make it more readable.

File: src/main/java/com/adyen/model/checkout/PaymentsRequest.java [hunk 2]
```diff
@@ -593,41 +581,29 @@ public void setOrderReference(String orderReference) {
...
     }
 
     public PaymentsRequest addEncryptedCardData(String encryptedCardNumber, String encryptedExpiryMonth, String encryptedExpiryYear, String encryptedSecurityCode, String holderName) {
-        this.paymentMethod = new HashMap<>();
-        this.paymentMethod.put(METHOD_TYPE, TYPE_SCHEME);
-        this.paymentMethod.put(ENCRYPTED_CARD_NUMBER, encryptedCardNumber);
-        this.paymentMethod.put(ENCRYPTED_EXPIRY_MONTH, encryptedExpiryMonth);
-        this.paymentMethod.put(ENCRYPTED_EXPIRY_YEAR, encryptedExpiryYear);
+        DefaultPaymentMethodDetails paymentMethodDetails = new DefaultPaymentMethodDetails();
+        this.paymentMethod = paymentMethodDetails;
```
Comment (rkewlani) [inline src/main/java/com/adyen/model/checkout/PaymentsRequest.java]: minor: exchange of 599 and 601 will make it more readable.

File: src/test/java/com/adyen/CheckoutTest.java [hunk 3]
```diff
@@ -196,6 +204,25 @@ public void TestPaymentMethodsFailureMissingIdentifierOnLive() throws Exception
 
     }
 
+    @Test
```
Comment (rkewlani) [inline src/test/java/com/adyen/CheckoutTest.java]: I think it will be a good idea to add one more test case for DefaultPaymentDetails class coverage.

Comment (lancergr) [reply src/test/java/com/adyen/CheckoutTest.java]: The intention here is to test the inject-ability of custom method details classes that implement PaymentMethodDetails. Generally we don't test model POJO classes and DefaultPaymentDetails is one of them, but feel free to send a suggestion of the extra test case if needed

Comment (lancergr) [reply src/test/java/com/adyen/CheckoutTest.java]: Testing PaymentMethodDetails is done at L211 and inject-ability of custom details class on L215.
Do you suggest to add one more test case or to remove the test case of L215 ? Could you give example?

Comment (rkewlani) [reply src/test/java/com/adyen/CheckoutTest.java]: If it is already covered in L211 then we are good to go :)

File: src/main/java/com/adyen/model/checkout/PaymentsRequest.java (hunk 1 above)
Comment (lancergr) [reply src/main/java/com/adyen/model/checkout/PaymentsRequest.java]: done

File: src/main/java/com/adyen/model/checkout/PaymentsRequest.java (hunk 2 above)
Comment (lancergr) [reply src/main/java/com/adyen/model/checkout/PaymentsRequest.java]: done

**Pull request merged**